### ISSUE TITLE
iOS support

### ISSUE THREAD

**Issue by dmitrykolesnikovich** [fabmax/kool#1]
Is there some technical nuance that makes iOS support undoable? Can I suggest my contribution if no?

**Comment by fabmax** [comment]
I myself have absoluetly no experience in iOS development, therefore I haven't looked into it so far. Obviously iOS doesn't run Java / kotlin directly, but it should be possible to to add a iOS target using kotlin-native.
Another thing is that OpenGL ES is deprecated on iOS and apps should use Metal as graphics API. As far as I know Metal is conceptually quite close to Vulkan (which I already implemented for desktop Java), so it should be possible to do this.

If you have experience with that stuff and / or wanna get into it, you are welcome do so!

**Comment by dmitrykolesnikovich** [comment]
I will provide iOS support by the end of the summer (hopefully). And based on that I really want to cooperate with you on gamedev platform that I created for myself as pet project (I have no solid understanding of 3d graphics but all the rest is pretty good). Do you consider such option interesting?

**Comment by fabmax** [comment]
In general this sounds interesting. However I probably won't be able to contribute much to an additional rather sophisticated project (due to time-constraints...). First, lets see how your iOS implementation goes, and then figure out what to do next.

**Comment by Ali-RS** [comment]
In case this helps, there is also this [ANGLE](https://github.com/google/angle) library being developed by Google that translates OpenGL ES calls to other APIs.

**Comment by fabmax** [comment]
ANGLE definitely looks interesting. Using it as a compatibility layer on iOS would solve a lot of problems. Lets hope they implement it soon.

**Comment by Ali-RS** [comment]
Yeah, that would be true for Mac OS as well if Apple decides to drop OpenGL completely in the future. Also with ANGLE Direct3D 11 backend, I guess it might be possible to target Xbox as well :)
